src/pymatgen/util/testing.py

```diff
@@ -0,0 +1,207 @@
+VASP_OUT_DIR: str = f"{TEST_FILES_DIR}/io/vasp/outputs"
+
+# Fake POTCARs have original header information, meaning properties like number of electrons,
+# nuclear charge, core radii, etc. are unchanged (important for testing) while values of the and
+# pseudopotential kinetic energy corrections are scrambled to avoid VASP copyright infringement
+FAKE_POTCAR_DIR: str = f"{VASP_IN_DIR}/fake_potcars"
+
+
+class PymatgenTest(TestCase):
+    """Extends unittest.TestCase with several convenient methods for testing:
+    - assert_msonable: Test if an object is MSONable and return the serialized object.
+    - assert_str_content_equal: Test if two string are equal (ignore whitespaces).
+    - get_structure: Load a Structure with its formula.
+    - serialize_with_pickle: Test if object(s) can be (de)serialized with `pickle`.
+    """
+
+    # dict of lazily-loaded test structures (initialized to None)
+    TEST_STRUCTURES: ClassVar[dict[PathLike, Structure | None]] = dict.fromkeys(STRUCTURES_DIR.glob("*"))
+
+    @pytest.fixture(autouse=True)
+    def _tmp_dir(self, tmp_path: Path, monkeypatch: pytest.MonkeyPatch) -> None:
+        """Make all tests run a in a temporary directory accessible via self.tmp_path.
+
+        References:
+            https://docs.pytest.org/en/stable/how-to/tmp_path.html
+        """
+        monkeypatch.chdir(tmp_path)  # change to temporary directory
+        self.tmp_path = tmp_path
+
+    @staticmethod
+    def assert_msonable(obj: Any, test_is_subclass: bool = True) -> str:
+        """Test if an object is MSONable and verify the contract is fulfilled,
+        and return the serialized object.
+
+        By default, the method tests whether obj is an instance of MSONable.
+        This check can be deactivated by setting `test_is_subclass` to False.
+
+        Args:
+            obj (Any): The object to be checked.
+            test_is_subclass (bool): Check if object is an instance of MSONable
+                or its subclasses.
+
+        Returns:
+            str: Serialized object.
+        """
+        obj_name = obj.__class__.__name__
+
+        # Check if is an instance of MONable (or its subclasses)
+        if test_is_subclass and not isinstance(obj, MSONable):
+            raise TypeError(f"{obj_name} object is not MSONable")
+
+        # Check if the object can be accurately reconstructed from its dict representation
+        if obj.as_dict() != type(obj).from_dict(obj.as_dict()).as_dict():
+            raise ValueError(f"{obj_name} object could not be reconstructed accurately from its dict representation.")
+
+        # Verify that the deserialized object's class is a subclass of the original object's class
+        json_str = json.dumps(obj.as_dict(), cls=MontyEncoder)
+        round_trip = json.loads(json_str, cls=MontyDecoder)
+        if not issubclass(type(round_trip), type(obj)):
+            raise TypeError(f"The reconstructed {round_trip.__class__.__name__} object is not a subclass of {obj_name}")
```

Comment:
This error message is more readable.

Before: `<class 'util.test_testing.TestAssertMSONable.test_not_round_trip.<locals>.NotAKpoints'> != <class 'pymatgen.io.vasp.inputs.Kpoints'>`

Now: `The reconstructed NotAKpoints object is not a subclass of Kpoints`